src/test/java/io/github/thebusybiscuit/slimefun4/utils/biomes/TestBiomeMapCompatibility.java

```diff
@@ -0,0 +1,169 @@
+import java.util.stream.Stream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+
+/**
+ * This test checks if biome maps work across multiple versions of Minecraft.
+ * A similar test can be written for Material Tags :?
+ * 
+ * @author TheBusyBiscuit
+ *
+ */
+class TestBiomeMapCompatibility {
+
+    private static final Map<MinecraftVersion, JsonArray> compatibilityMap = new EnumMap<>(MinecraftVersion.class);
+
+    @BeforeAll
+    public static void load() {
+        MockBukkit.mock();
+
+        for (MinecraftVersion version : MinecraftVersion.values()) {
+            if (!version.isVirtual()) {
+                loadBiomes(version);
+            }
+        }
+    }
+
+    @AfterAll
+    public static void unload() {
+        MockBukkit.unmock();
+    }
+
+    private static void loadBiomes(@Nonnull MinecraftVersion version) {
+        String path = "/biomes/" + version.getName() + ".json";
+
+        if (Slimefun.class.getResource(path) == null) {
+            return;
+        }
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(Slimefun.class.getResourceAsStream(path), StandardCharsets.UTF_8))) {
+            JsonParser parser = new JsonParser();
+            JsonArray biomes = parser.parse(reader.lines().collect(Collectors.joining(""))).getAsJsonArray();
+
+            compatibilityMap.put(version, biomes);
+        } catch (IOException x) {
+            x.printStackTrace();
+        }
+    }
+
+    @ParameterizedTest(name = "Test if {0} has a biome resource")
+    @EnumSource(value = MinecraftVersion.class)
+    void testMissingBiomeResources(@Nonnull MinecraftVersion version) {
+        if (!version.isVirtual()) {
+            Assertions.assertTrue(compatibilityMap.containsKey(version));
+        }
+    }
+
+    @ParameterizedTest(name = "Test if biome-map {0}.json is compatible with {1}")
+    @MethodSource("biomeMaps")
+    @ParametersAreNonnullByDefault
+    void testCompatibilities(String name, MinecraftVersion version) {
+        assertCompatibility(name, version);
+    }
+
+    /**
+     * We manually specify here which biome map works on which minecraft versions.
+     * 
+     * @return A {@link Stream} of {@link Arguments} for our unit test.
+     */
+    private static @Nonnull Stream<Arguments> biomeMaps() {
```

Comment:
This method is currently unused, what's the purpose here?

Reply:
Look above. And look at the method.
It's a `Stream<Arguments>` method.
It's used for parameterized unit tests.

Reply:
Oh whoops, I didn't realize you could do MethodSource and specify a name... huh neat.

Reply:
Yup, it is very neat!
I actually use it relatively often: https://github.com/Slimefun/Slimefun4/search?q=MethodSource